rpc/src/v1/impls/ibc.rs

```diff
@@ -102,28 +102,33 @@ where
     }
 
     fn compose_header(&self, block_number: u64) -> Result<Option<Bytes>> {
+        // FIXME: Currently compose_header() works only for at most bestblock-1
+        let block_id_prev = BlockId::Number(block_number - 1);
         let block_id = BlockId::Number(block_number);
+        let block_id_next = BlockId::Number(block_number + 1);
+        if self.client.state_at(block_id_prev).is_none() {
+            return Ok(None)
+        }
         if self.client.state_at(block_id).is_none() {
             return Ok(None)
         }
-        let state = self.client.state_at(block_id).unwrap();
-
+        if self.client.state_at(block_id_next).is_none() {
+            return Ok(None)
+        }
+        let state_prev = self.client.state_at(block_id_prev).unwrap();
         let header_core = self.client.block_header(&block_id).unwrap();
-        let vset_raw = ccore::stake::NextValidators::load_from_state(&state).map_err(errors::core)?;
+        let header_core_next = self.client.block_header(&block_id_next).unwrap();
+        let vset_raw = ccore::stake::NextValidators::load_from_state(&state_prev).map_err(errors::core)?;
```

Comment:
If we use `CurrentValidators`, we don't need to read the previous block.

Reply:
Ok, then I'll add `create_compact_validator_set()` on `CurrentValidators` and use that.